### ISSUE TITLE
Embeddables: `fieldName` option is ignored

### ISSUE THREAD

**Issue by edorgeville** [mikro-orm/mikro-orm#2165]
**Describe the bug**
Now that #1948 is implemented and merged (thank you so much 🙌) I am trying to migrate some code that was previously using `ObjectId`s to embeddables + `m:1`. 
It seems that the `fieldName` parameter in `ManyToOne` is ignored.

**To Reproduce**

<details>
  <summary>Code</summary>

```typescript
import { Embeddable, Embedded, Entity, ManyToOne, MikroORM, PrimaryKey, Property } from '@mikro-orm/core';
import type { SqliteDriver } from '@mikro-orm/sqlite';

@Entity()
export class User {

  @PrimaryKey()
  id!: number;

  @Property()
  name: string = '';

  constructor(name: string) {
    this.name = name;
  }

}

@Embeddable()
export class FamilyMember {

  @Property()
  relation: string = 'brother';

  @ManyToOne(() => User, { eager: true, fieldName: 'user_id' }) // we want to save our relationship inside a `user_id` field
  user!: User;

}

@Entity()
export class Family {

  @PrimaryKey()
  id!: number;

  @Embedded(() => FamilyMember, { array: true })
  members: FamilyMember[] = [];

}

describe('GH issue 1003', () => {

  let orm: MikroORM<SqliteDriver>;

  beforeAll(async () => {
    orm = await MikroORM.init({
      entities: [FamilyMember, User, Family],
      dbName: ':memory:',
      type: 'sqlite',
    });
    await orm.getSchemaGenerator().createSchema();
  });

  afterAll(async () => {
    await orm.close(true);
  });

  test(`GH issue 2165`, async () => {
    const family = new Family();

    const dad = new FamilyMember();
    dad.relation = 'dad';
    dad.user = new User('John');
    family.members.push(dad);

    const mom = new FamilyMember();
    mom.relation = 'mom';
    mom.user = new User('Jane');
    family.members.push(mom);

    console.log(family);

    await orm.em.persistAndFlush(family);

    const nativeResults: Family[] = await orm.em.createQueryBuilder(Family).execute();

    // Relationship is saved in `user` field instead of `user_id`
    console.log(nativeResults[0].members);
  });

});
```
</details>

Also available in repro-repo here: https://github.com/edorgeville/mikro-orm-embeddable-m1-fieldName

**Expected behavior**
Expected relationships to be saved according to `fieldName` property. Instead, it uses the class property name.

**Versions**

| Dependency | Version |
| - | - |
| node | v14.17.3 |
| typescript | v10.0.0 |
| @mikro-orm/core | ^5.0.0-dev.250 |
| @mikro-orm/mongodb | ^5.0.0-dev.250 |

**Comment by titivuk** [comment]
Hello
Any updates on this?

**Comment by B4nan** [comment]
Closing as fixed in v6 via #4866
